### DISCUSSION TITLE
GSI examples - Fortran wrapper

### DISCUSSION THREAD

**Discussion by lcutrone** [mutationpp/Mutationpp#157] [Q&A] [answered]
Hi, 
are they available additional examples for the fortran wrapper, and in particular for the gsi module?
I would like to use mutation++ to solve surface mass and energy balance for an ablative boundary condition to be implemented in my own CFD solver.

Thanks,
Luigi

**Comment by capriatim** [comment, marked as answer]
Hi Luigi,

you should first of all create a GSI.xml file to specify simulation surface reactions. You can find some examples in _Mutationpp/tests/data/gsi_.
Then you have to set the problem up in your CFD solver:
**mass balance:**
            CALL mpp_set_surface_state( partial density on the wall , Twall)
            CALL mpp_set_diffusion_model( mole fractions of the first cell close to the surface, distance cell center - wall )
            CALL mpp_solve_surface_balance()
            CALL mpp_get_surface_state(partial density on the wall, Twall)
            CALL mpp_mass_blowing_rate(mdot)
            
**mass-energy balance:**
same story, but you have to add the gas conductive heat flux to the wall
        CALL mpp_set_cond_heat_flux( T of the first cell close to the surface, distance cell center wall )
        
Michele
